### DISCUSSION TITLE
How to specify a sidebar in _metadata.yml?

### DISCUSSION THREAD

**Discussion by fredguth** [quarto-dev/quarto-cli#9557] [Q&A]
### Description

I would like to create a website that works like a *portal* (meaning aggregating subsites).
The portal has sections like 'Datasets', 'Dashboards', .... and each section of the portal lists `index.qmd` files inside their directories. Neither the portal nor the listing sections have sidebars. But when I select a link in the listing, I go to a `subsite`, somewhere like 'Datasets/my-latest-project/index.qmd' with its own sidebar.
 
I *think* I  need a sidebar instead of a TOC navigation because each subpage of the subsite has its own toc and shouldn't be seen from the subsite/index.qmd TOC.

I wanted to specify the sidebar contents locally in each subsite subdirectory.  Using `quarto-web` as an example, each section will be a listing like the Blog section and each link will direct to a subsite that will be similar to the `References` section. 

Reading `quarto-web` code, I notice the `References` section has its sidebar defined in the root _quarto.yml and *again* in a _reference.yml which I believe is used only for automatically creating links in its /docs/reference/index.qmd page. 

Isn't there a way to add the sidebar information in a subdirectory `_metadata.yml` file? Is there an example of something like what I am looking for?

---
````

Quarto 99.9.9
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      commit: 2d068c35a2d5040a515fed1b6e991514144245f6
      Path: /Users/fredguth/Code/quarto-cli/package/dist/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.01
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/fredguth/Library/TinyTeX/bin/universal-darwin
      Version: 2022

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.7
      Path: /opt/homebrew/opt/python@3.11/bin/python3.11
      Jupyter: 5.3.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.2
      Path: /opt/homebrew/Cellar/r/4.3.2/lib/R
      LibPaths:
        - /opt/homebrew/lib/R/4.3/site-library
        - /opt/homebrew/Cellar/r/4.3.2/lib/R/library
      knitr: 1.46
      rmarkdown: 2.26

[✓] Checking Knitr engine render......OK
````

**Comment by mcanouil** [comment]
You need to define all the sidebars in `_quarto.yml`. Add (meaningful) IDs to them.
Then add `sidebar: meaningful-id` in `_metadata.yml` for each of your sub-directories.

**Reply by fredguth** [reply]
So I need to change the root _quarto.yml everytime, right?  There is no way to compose the sidebar in the same way _metadata.yml is merged, right? No, _sidebar.yml

**Reply by mcanouil** [reply]
I don't know what you mean by "everytime".
If you manually add document and title for every entry that's a choice.
In this case, I use `contents: directory` and use the `title` of each document automatically, so I never have to change the sidebar. Downside, you don't get sections.
It's possible (far from certain) that `contents` accepts a YAML file which you could have in your subdirectories.
